pandas/core/internals/construction.py

```diff
@@ -74,6 +77,8 @@ def arrays_to_mgr(arrays, arr_names, index, columns, dtype=None, verify_integrit
     # from BlockManager perspective
     axes = [columns, index]
 
+    assert isinstance(columns, Index), type(columns)
+    assert isinstance(index, Index), type(index)
```

Comment:
ensure_index on L70 and L75 should return an Index. can you type ensure_index instead of adding asserts